### PR TITLE
Show error when no documentation

### LINKED ISSUE #33: Show error message when documentation is not found
When creating a new resource, if the kind is not set up yet (or is the wrong kind), the documentation tab should show an error message instead of keeping the spinner showing forever.

**Acceptance Criteria:**
- [ ] When the resource has a kind and documentation is found, the documentation is shown
- [ ] If there is an error fetching the documentation, it should be shown in red with a message like: `Cannot load documentation: CAUSE HERE`.

### PR DESCRIPTION
**Pull request by ashu8912**
fixes #33

### CODE REVIEW

File: frontend/src/components/common/EditorDialog.tsx [hunk 1]
```diff
@@ -292,16 +292,28 @@ const useStyles = makeStyles(theme => ({
...
+        `);
       return;
     }
 
     getDocDefinitions(docSpecs.apiVersion, docSpecs.kind)
       .then(result => {
         setDocs(result?.properties || {});
+      })
+      .catch((err) => {
+        setDocsError(`${docsErrorCommonString}: ${err}`);
```
Comment (joaquimrocha) [inline frontend/src/components/common/EditorDialog.tsx:316]: This one will end up with 2 `:` in the message.

File: frontend/src/components/common/EditorDialog.tsx [hunk 2]
```diff
@@ -292,16 +292,28 @@ const useStyles = makeStyles(theme => ({
 function DocsViewer(props: {docSpecs: any}) {
   const { docSpecs } = props;
   const classes = useStyles();
+  const docsErrorCommonString = 'Cannot load documentation:';
```
Comment (joaquimrocha) [inline frontend/src/components/common/EditorDialog.tsx:295]: I understand why you used a var here, but this is a message and will likely get translated later. As a rule one shouldn't do things like concatenation of translatable strings programatically like this patch does, so maybe just repeat the text.

File: frontend/src/components/common/EditorDialog.tsx [hunk 3]
```diff
@@ -333,13 +345,10 @@ function DocsViewer(props: {docSpecs: any}) {
...
-      <Empty>
-        Error getting documentation!
-        Please make sure the YAML is valid and has the kind and apiVersion set.
-      </Empty>
-      : docs === null ?
-        <Loader />
+    docs === null && docsError === null ?
+      <Loader />
+      : !_.isEmpty(docsError) ?
+        <Empty><Typography color="secondary" align="center">{docsError}</Typography></Empty>
```
Comment (joaquimrocha) [inline frontend/src/components/common/EditorDialog.tsx:351]: Hmm, you shouldn't need the `typography/secondary` elem here, right? Should all be in the `<Empty>` component.

Comment (ashu8912) [reply frontend/src/components/common/EditorDialog.tsx:351]: @joaquimrocha i added the Typography inside Empty element because right now our Empty Component doesn't have the facility to pass color from outside because its hardcoded there.And in this case I need to pass the red color in case of errors

Comment (joaquimrocha) [reply frontend/src/components/common/EditorDialog.tsx:351]: Ah, but you're using the secondary color. I think you shoud use the `error` color.
Maybe a quick patch just passing the color down to the empty would be better than wrapping this component.

File: frontend/src/components/common/EditorDialog.tsx [hunk 4]
```diff
@@ -293,15 +293,26 @@ function DocsViewer(props: {docSpecs: any}) {
   const { docSpecs } = props;
   const classes = useStyles();
   const [docs, setDocs] = React.useState<object | null>(null);
-
```
Comment (joaquimrocha) [inline frontend/src/components/common/EditorDialog.tsx:296]: Please don't remove newlines like this. You should add your new state line to the top of the newline.

File: frontend/src/components/common/EditorDialog.tsx [hunk 5]
```diff
@@ -293,15 +293,26 @@ function DocsViewer(props: {docSpecs: any}) {
...
   React.useEffect(() => {
+    if (docSpecs.error) {
+      setDocsError(`Cannot load documentation: ${docSpecs.error}`);
+      return;
+    }
     if (!docSpecs.apiVersion || !docSpecs.kind) {
+      setDocsError(
+        `Cannot load documentation: 
+        Please make sure the YAML is valid and has the kind and apiVersion set.
+        `);
```
Comment (joaquimrocha) [inline frontend/src/components/common/EditorDialog.tsx:306]: Nitpick: Are you using the backquotes here for preserving the space or was it a mistake? You can use a longer than 100 chars line if it's a string (with single quotes).

File: frontend/src/components/common/EditorDialog.tsx [hunk 6]
```diff
@@ -333,13 +344,10 @@ function DocsViewer(props: {docSpecs: any}) {
...
-    docSpecs.error ?
-      <Empty>
-        Error getting documentation!
-        Please make sure the YAML is valid and has the kind and apiVersion set.
-      </Empty>
-      : docs === null ?
-        <Loader />
+    docs === null && docsError === null ?
+      <Loader />
+      : !_.isEmpty(docsError) ?
```
Comment (joaquimrocha) [inline frontend/src/components/common/EditorDialog.tsx:349]: Nitpick: You can easily just check for the `docsErorr` without using the `isEmpty` (as will only be valid if it's a non-empty string here).
But you don't really have to change this, it's just for future reference.

File: frontend/src/components/common/EmptyContent.tsx [hunk 7]
```diff
@@ -1,17 +1,19 @@
 import Box from '@material-ui/core/Box';
-import Typography from '@material-ui/core/Typography';
+import Typography, { TypographyProps } from '@material-ui/core/Typography';
 import React from 'react';
 
-export default function Empty(props: React.PropsWithChildren<{}>) {
+export default function Empty(props: {color: TypographyProps['color'];children: React.ReactNode}) {
```
Comment (joaquimrocha) [inline frontend/src/components/common/EmptyContent.tsx:5]: Can't you use?
```suggestion
export default function Empty(props: React.PropsWithChildren<{color: TypographyProps['color']}>) {
```

File: frontend/src/components/common/EditorDialog.tsx [hunk 8]
```diff
@@ -347,7 +347,7 @@ function DocsViewer(props: {docSpecs: any}) {
     docs === null && docsError === null ?
       <Loader />
       : !_.isEmpty(docsError) ?
-        <Empty>{docsError}</Empty>
+        <Empty color="error">{docsError}</Empty>
```
Comment (joaquimrocha) [inline frontend/src/components/common/EditorDialog.tsx:350]: BTW, this change shouldn't be part of this commit.
